### PR TITLE
[Part3] Implement test files

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This is the third part of unit tests implementation, following and implemented on top of Part2.

## Changes
1. test files: implement test files for `boot_control` module.
2. github action: use `MishaKav/pytest-coverage-comment@v1.1.32` to record the coverage report in github action.
3. test files: use `TestConfiguration` in `conftest.py` to store configuration for the whole test sessions.
4. test files: refine the `ota-test_base` image Dockerfile to reduce the generated image size
5. grub, configs: introduce new `ACTIVE_ROOTFS_PATH` option, and integrate it into the grub controller module.
6. test files: other minor updates to already implemented test files.

### CODE REVIEW

**Comment by Bodong-Yang** [comment]
A bug represented as the github action CI pipeline doesn't fail on pytest failed, I will fix it now.

File: tests/conftest.py [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
+    OTA_IMAGE_DIR = "/ota-image"
+    OTA_IMAGE_SERVER_ADDR = "127.0.0.1"
+    OTA_IMAGE_SERVER_PORT = 8080
+    KERNEL_VERSION = "5.8.0-53-generic"
+    CURRENT_VERSION = "123.x"
+    UPDATE_VERSION = "789.x"
 
+    # slots settings for testing
+    SLOT_A_UUID = "aaaaaaaa-0000-0000-0000-aaaaaaaaaaaa"
+    SLOT_A_PARTUUID = SLOT_A_UUID
```
Comment (obi-t4) [inline tests/conftest.py:25]: In general, UUID and PARTUUID are different values.

File: tests/test_boot_control/default_grub [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Uncomment if you don't want GRUB to pass "root=UUID=xxx" parameter to Linux
+#GRUB_DISABLE_LINUX_UUID=true
+
+# Uncomment to disable generation of recovery mode menu entries
+#GRUB_DISABLE_RECOVERY="true"
+
+# Uncomment to get a beep at grub start
+#GRUB_INIT_TUNE="480 440 1"
+GRUB_DISABLE_SUBMENU=y
+GRUB_DISABLE_OS_PROBER=true
```
Comment (obi-t4) [inline tests/test_boot_control/default_grub:35]: In general, the last line should contain LF (=0x0a). most of the cases /etc/default/grub also contain LF.

File: tests/test_boot_control/extlinux.conf_slot_a [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#    LINUX /boot/Image.backup
+#    INITRD /boot/initrd
+#    APPEND ${cbootargs}
+
+LABEL Tier4 ISX021 GMSL2 Camera Device Tree Overlay
+	MENU LABEL Tier4 ISX021 GMSL2 Camera Device Tree Overlay
+	LINUX /boot/Image
+	FDT /boot/kernel_tegra194-rqx-58g-tier4-isx021-gmsl2-camera-device-tree-overlay.dtb
+	INITRD /boot/initrd
+	APPEND ${cbootargs} quiet root=PARTUUID=aaaaaaaa-0000-0000-0000-aaaaaaaaaaaa rw rootwait rootfstype=ext4 console=ttyTCU0,115200n8 console=tty0 fbcon=map:0 net.ifnames=0 rootfstype=ext4
```
Comment (obi-t4) [inline tests/test_boot_control/extlinux.conf_slot_a:37]: /boot/extlinux/extlinux.conf also contains LF in the last line in general.

File: tests/test_boot_control/fstab_origin [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# <file system> <mount point>   <type>  <options>       <dump>  <pass>
+# / was on /dev/sda2 during installation
+UUID=44d14df0-31be-4fe2-b05a-bae6cf157ea4	/	ext4	errors=remount-ro	0	1
+# /boot was on /dev/sda1 during installation
+UUID=ca573144-b0a4-44f6-a750-45a169c3c6e7	/boot	ext4	defaults	0	2
+/swapfile	none	swap	sw	0	0
+tmpfs	/media/autoware/LOG	tmpfs	rw,nosuid,nodev,noexec,nofail,size=10G,mode=1755	0	0
+LABEL=ROSBAG	/media/autoware/ROSBAG	ext4	nofail	0	0
+tmpfs	/mnt/LOG	tmpfs	rw,nosuid,nodev,noexec,nofail,size=10G,mode=1755	0	0
+LABEL=ROSBAG	/mnt/ROSBAG	ext4	nofail	0	0
```
Comment (obi-t4) [inline tests/test_boot_control/fstab_origin:16]: should contain LF in the last line.

File: tests/test_boot_control/fstab_updated [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# <file system> <mount point>   <type>  <options>       <dump>  <pass>
+# / was on /dev/sda2 during installation
+UUID=bbbbbbbb-1111-1111-1111-bbbbbbbbbbbb	/	ext4	errors=remount-ro	0	1
+# /boot was on /dev/sda1 during installation
+UUID=ca573144-b0a4-44f6-a750-45a169c3c6e7	/boot	ext4	defaults	0	2
+/swapfile	none	swap	sw	0	0
+tmpfs	/media/autoware/LOG	tmpfs	rw,nosuid,nodev,noexec,nofail,size=10G,mode=1755	0	0
+LABEL=ROSBAG	/media/autoware/ROSBAG	ext4	nofail	0	0
+tmpfs	/mnt/LOG	tmpfs	rw,nosuid,nodev,noexec,nofail,size=10G,mode=1755	0	0
+LABEL=ROSBAG	/mnt/ROSBAG	ext4	nofail	0	0
```
Comment (obi-t4) [inline tests/test_boot_control/fstab_updated:16]: should contain LF in the last line.

**Review by obi-t4 — COMMENTED** [review]
I left some comments.

File: tests/conftest.py (hunk 1 above)
Comment (Bodong-Yang) [reply tests/conftest.py:25]: Yes, use the same value is only for test convenience, `grub` use `UUID` and `cboot` use `PARTUUID`. So I use different var name here.

Comment (obi-t4) [reply tests/conftest.py:25]: Could you add the comment, why UUID and PARTUUID can be the same value?

Comment (Bodong-Yang) [reply tests/conftest.py:25]: Sure!

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**